controllers/fluentdconfig_controller.go

```diff
@@ -211,25 +210,22 @@ func (r *FluentdConfigReconciler) ClusterCfgsForFluentd(
 		return err
 	}
 
-	allNamespaces := make([]string, 0)
-
 	for _, cfg := range clustercfgs.Items {
 		// If the field watchedNamespaces is empty, all namesapces will be watched.
 		watchedNamespaces := cfg.GetWatchedNamespaces()
 
 		if len(watchedNamespaces) == 0 {
-			if len(allNamespaces) == 0 {
-				var namespaceList corev1.NamespaceList
-				if err := r.List(ctx, &namespaceList); err != nil {
-					return err
-				}
-
-				for _, item := range namespaceList.Items {
-					allNamespaces = append(allNamespaces, item.Name)
-				}
+			var namespaceList corev1.NamespaceList
+			if err := r.List(ctx, &namespaceList); err != nil {
+				return err
+			}
+
+			for _, item := range namespaceList.Items {
+				watchedNamespaces = append(watchedNamespaces, item.Name)
 			}
 
-			cfg.Spec.WatchedNamespaces = allNamespaces
+			// Don't patch the CR, or it would requeue.
+			cfg.Spec.WatchedNamespaces = watchedNamespaces
 		}
```

Comment:
Do we have to put every namespace into WatchedNamespaces?
No easier way to represent <all namespace> instead of iterating every namespace?

Reply:
The former codes cache it, I wonder if the namespaces changed, would mix some namespaces.

Reply:
Let's discuss this later, I think there is no need to iterating every ns in case of all namespace

Reply:
OK.